File: app/src/main/java/com/evan/kotlintest/AndroidIdUtils.java
package com.evan.kotlintest;

import android.content.Context;
import android.os.Environment;
import android.provider.Settings;

import java.io.*;

public class AndroidIdUtils {
    private static final String filePath = Environment.getExternalStorageDirectory().toString() + "/jmad/android_id.data";
    private static final String idSpliterator = "#_#";

    public static String getAndroidId(Context context, String prdId) {
        File file = new File(filePath);
        String id = null;
        try {
            if (file.exists()) {
                BufferedReader br = new BufferedReader(new FileReader(file));
                for (; ; ) {
                    id = br.readLine();
                    if (id == null) break;
                    if (id.startsWith(prdId)) {
                        return id.split(idSpliterator)[1];
                    }
                }
                br.close();
            } else {
                file.getParentFile().mkdirs();
                file.createNewFile();
            }
            id = Settings.System.getString(context.getContentResolver(), Settings.System.ANDROID_ID);
            BufferedWriter bw = new BufferedWriter(new FileWriter(file, true));
            bw.newLine();
            bw.write(prdId + idSpliterator + id);
            bw.flush();
            bw.close();
        } catch (Exception e) {
        }
        return id;
    }
}

File: app/src/main/java/com/evan/kotlintest/AndroidIDUtils.kt
package com.evan.kotlintest

import android.content.Context
import android.os.Environment
import android.provider.Settings
import java.io.*

object AndroidIDUtils {
    private val filePath = Environment.getExternalStorageDirectory().toString() + "/jmad/android_id.data"
    private val idSpliterator = "#_#"

    fun getAndroidId(context: Context, prdId: String): String {
        val file = File(filePath)
        if (file.exists()) {
            val br = BufferedReader(FileReader(file))
            while (true) {
                val line = br.readLine() ?: break
                if (line.startsWith(prdId)) {
                    return line.split(idSpliterator)[1]
                }
            }
            br.close()
        } else {
            file.parentFile.mkdirs()
            file.createNewFile()
        }
        val id = Settings.System.getString(context.getContentResolver(), Settings.System.ANDROID_ID);
        val bw = BufferedWriter(FileWriter(file, true))
        bw.newLine()
        bw.write("$prdId$idSpliterator$id")
        bw.flush()
        bw.close()
        return id
    }

}